# Pakistan's Job Market

## Importing Libraries

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

### Q1 Read the Dataset and Explore the dataset by checking shape, and  columns, and see the first/last rows using head/tail. (n= 5,15,30)

In [2]:
df = pd.read_csv('Pakistan Available Job.csv')
df

Job Name        label  \
0     Full Time New Job Positions .Net, .Netcore, Fl...  Premium Job   
1       Full Time Senior Web Developer Jobs in Pakistan  Premium Job   
2           Full Time Russian Speakers Jobs in Pakistan  Premium Job   
3     Full Time Customer Support Specialist - Intern...  Premium Job   
4     Full Time English Speaker - International Busi...  Premium Job   
...                                                 ...          ...   
6675  Full Time Senior Software Engineer  Job in Pak...          NaN   
6676  Full Time Commercial Experience Executive Job ...          NaN   
6677  Full Time Business Development Executive Job i...          NaN   
6678  Full Time 3D Modeler / CG Artist Game Jobs in ...          NaN   
6679  Full Time Bidding Expert / Social Media Market...          NaN   

                                 Company Name        Job Type  \
0                   Nayel Solutions, Pakistan  Full Time Jobs   
1     Eurosoft Tech Private Limited, Pakistan  Full Time Jobs   
2                         ICM JAPAN, Pakistan  Full Time Jobs   
3                              ibex, Pakistan  Full Time Jobs   
4                         ICM JAPAN, Pakistan   Full Time Job   
...                                       ...             ...   
6675                       KnovaSol, Pakistan   Full Time Job   
6676                                      NaN   Full Time Job   
6677                  Loop Brackets, Pakistan   Full Time Job   
6678            Super Duper Studio , Pakistan  Full Time Jobs   
6679            Super Duper Studio , Pakistan  Full Time Jobs   

          Experience Required              Department  \
0            2 Years Job Exp.                 IT Jobs   
1            2 Years Job Exp.                 IT Jobs   
2                    < 1 Year   Customer Service Jobs   
3     Job for Fresh Graduates   Customer Service Jobs   
4                    < 1 Year    Customer Service Job   
...                       ...                     ...   
6675         3 Years Job Exp.   Computer Software Job   
6676         2 Years Job Exp.               Admin Job   
6677         2 Years Job Exp.   Computer Software Job   
6678         2 Years Job Exp.  Computer Software Jobs   
6679  Job for Fresh Graduates  Computer Software Jobs   

                                                     JD       City Date Posted  
0     New Job Positions .net, .netcore, flutter, Tea...  Islamabad   12-Mar-21  
1     We are looking for an experienced Web Develope...    Karachi   12-Mar-21  
2     International clients dealing exposure (B2B).S...    Karachi   12-Mar-21  
3     Responsible for acting as a liaison between ou...  Islamabad    9-Mar-21  
4     International clients dealing exposure (B2B) a...    Karachi    5-Mar-21  
...                                                 ...        ...         ...  
6675  We required services of Senior Software Engine...  Islamabad   30-Dec-20  
6676  As one of the leading employers in the country...     Multan   29-Dec-20  
6677  The ideal candidate will have experience in al...     Lahore   29-Dec-20  
6678  Must be able to create 3D Game Environments an...     Lahore   29-Dec-20  
6679  Candidate Shall Be Expert in Getting Work from...     Lahore   29-Dec-20  

[6680 rows x 9 columns]

In [22]:
type(df)

pandas.core.frame.DataFrame

In [5]:
# shape
df.shape

(6680, 9)

In [6]:
# Head
df.head(5)

Job Name        label  \
0  Full Time New Job Positions .Net, .Netcore, Fl...  Premium Job   
1    Full Time Senior Web Developer Jobs in Pakistan  Premium Job   
2        Full Time Russian Speakers Jobs in Pakistan  Premium Job   
3  Full Time Customer Support Specialist - Intern...  Premium Job   
4  Full Time English Speaker - International Busi...  Premium Job   

                              Company Name        Job Type  \
0                Nayel Solutions, Pakistan  Full Time Jobs   
1  Eurosoft Tech Private Limited, Pakistan  Full Time Jobs   
2                      ICM JAPAN, Pakistan  Full Time Jobs   
3                           ibex, Pakistan  Full Time Jobs   
4                      ICM JAPAN, Pakistan   Full Time Job   

       Experience Required             Department  \
0         2 Years Job Exp.                IT Jobs   
1         2 Years Job Exp.                IT Jobs   
2                 < 1 Year  Customer Service Jobs   
3  Job for Fresh Graduates  Customer Service Jobs   
4                 < 1 Year   Customer Service Job   

                                                  JD       City Date Posted  
0  New Job Positions .net, .netcore, flutter, Tea...  Islamabad   12-Mar-21  
1  We are looking for an experienced Web Develope...    Karachi   12-Mar-21  
2  International clients dealing exposure (B2B).S...    Karachi   12-Mar-21  
3  Responsible for acting as a liaison between ou...  Islamabad    9-Mar-21  
4  International clients dealing exposure (B2B) a...    Karachi    5-Mar-21

In [7]:
# Tail
df.tail(10)

Job Name label  \
6670  Full Time Assistant Manager / Manager Sales Jo...   NaN   
6671  Full Time Sales / Customer Service Executive  ...   NaN   
6672          Full Time Sales Executive Job in Pakistan   NaN   
6673       Full Time Call Center Agent Jobs in Pakistan   NaN   
6674  Full Time Senior Python Developer  Jobs in Pak...   NaN   
6675  Full Time Senior Software Engineer  Job in Pak...   NaN   
6676  Full Time Commercial Experience Executive Job ...   NaN   
6677  Full Time Business Development Executive Job i...   NaN   
6678  Full Time 3D Modeler / CG Artist Game Jobs in ...   NaN   
6679  Full Time Bidding Expert / Social Media Market...   NaN   

                                Company Name        Job Type  \
6670         Honda South (Pvt) Ltd, Pakistan  Full Time Jobs   
6671         Honda South (Pvt) Ltd, Pakistan   Full Time Job   
6672         Honda South (Pvt) Ltd, Pakistan   Full Time Job   
6673  Vents Marketing & Management, Pakistan  Full Time Jobs   
6674  Bridging Trade International, Pakistan  Full Time Jobs   
6675                      KnovaSol, Pakistan   Full Time Job   
6676                                     NaN   Full Time Job   
6677                 Loop Brackets, Pakistan   Full Time Job   
6678           Super Duper Studio , Pakistan  Full Time Jobs   
6679           Super Duper Studio , Pakistan  Full Time Jobs   

          Experience Required              Department  \
6670                 < 1 Year              Autos Jobs   
6671         3 Years Job Exp.               Autos Job   
6672         3 Years Job Exp.               Autos Job   
6673          1 Year Job Exp.   Customer Service Jobs   
6674         4 Years Job Exp.                 IT Jobs   
6675         3 Years Job Exp.   Computer Software Job   
6676         2 Years Job Exp.               Admin Job   
6677         2 Years Job Exp.   Computer Software Job   
6678         2 Years Job Exp.  Computer Software Jobs   
6679  Job for Fresh Graduates  Computer Software Jobs   

                                                     JD       City Date Posted  
6670  To assist the head of sales.Lead / Manage the ...    Karachi   30-Dec-20  
6671  Calls and visits to Corporates &amp; Banks.Coo...    Karachi   30-Dec-20  
6672  Honda dealership sales.Increase the sales numb...    Karachi   30-Dec-20  
6673  We are hiring Call Center agents for Internati...    Karachi   30-Dec-20  
6674  What they will do:&middot; Development of appl...  Islamabad   30-Dec-20  
6675  We required services of Senior Software Engine...  Islamabad   30-Dec-20  
6676  As one of the leading employers in the country...     Multan   29-Dec-20  
6677  The ideal candidate will have experience in al...     Lahore   29-Dec-20  
6678  Must be able to create 3D Game Environments an...     Lahore   29-Dec-20  
6679  Candidate Shall Be Expert in Getting Work from...     Lahore   29-Dec-20

### Q2 Extract and document key insights by exploring column data types and applying info and describe and write down your insights inside the markdown

**This Dataset has 8 Columns & 6680 Rows <br>
The datatype of each column is Object**

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6680 entries, 0 to 6679
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Job Name             6680 non-null   object
 1   label                1115 non-null   object
 2   Company Name         6017 non-null   object
 3   Job Type             6680 non-null   object
 4   Experience Required  6680 non-null   object
 5   Department           6680 non-null   object
 6   JD                   6680 non-null   object
 7   City                 6680 non-null   object
 8   Date Posted          6680 non-null   object
dtypes: object(9)
memory usage: 469.8+ KB


**Count:** is the number of values in that column.<br>
**Unique:** the number of unique values in that column<br>
**Top:** Most common value in that column<br>
**Freq:** is the most common value's frequency

In [15]:
df.describe()

Job Name        label  \
count                                         6680         1115   
unique                                        3838            3   
top     Full Time Graphic Designer Job in Pakistan  Gallery Job   
freq                                            60          775   

          Company Name       Job Type Experience Required Department  \
count             6017           6680                6680       6680   
unique            2879              2                  17        111   
top     ibex, Pakistan  Full Time Job    2 Years Job Exp.     IT Job   
freq                93           3821                1635        820   

                                                       JD    City Date Posted  
count                                                6680    6680        6680  
unique                                               5043      86         443  
top     To generate quality leads that result in sales...  Lahore   23-Feb-21  
freq                                                   37    2046         134

### Q3 Filter numeric columns and summarize their statistics and do the same for categorical columns

Because the dataset has only object datatype

In [21]:
df.describe(include="object")

Job Name        label  \
count                                         6680         1115   
unique                                        3838            3   
top     Full Time Graphic Designer Job in Pakistan  Gallery Job   
freq                                            60          775   

          Company Name       Job Type Experience Required Department  \
count             6017           6680                6680       6680   
unique            2879              2                  17        111   
top     ibex, Pakistan  Full Time Job    2 Years Job Exp.     IT Job   
freq                93           3821                1635        820   

                                                       JD    City Date Posted  
count                                                6680    6680        6680  
unique                                               5043      86         443  
top     To generate quality leads that result in sales...  Lahore   23-Feb-21  
freq                                                   37    2046         134

**Created a variable df_obj and stored all the object datatype columns in it**

In [28]:
df_obj = df.select_dtypes(["object"])
df_obj

Job Name        label  \
0     Full Time New Job Positions .Net, .Netcore, Fl...  Premium Job   
1       Full Time Senior Web Developer Jobs in Pakistan  Premium Job   
2           Full Time Russian Speakers Jobs in Pakistan  Premium Job   
3     Full Time Customer Support Specialist - Intern...  Premium Job   
4     Full Time English Speaker - International Busi...  Premium Job   
...                                                 ...          ...   
6675  Full Time Senior Software Engineer  Job in Pak...          NaN   
6676  Full Time Commercial Experience Executive Job ...          NaN   
6677  Full Time Business Development Executive Job i...          NaN   
6678  Full Time 3D Modeler / CG Artist Game Jobs in ...          NaN   
6679  Full Time Bidding Expert / Social Media Market...          NaN   

                                 Company Name        Job Type  \
0                   Nayel Solutions, Pakistan  Full Time Jobs   
1     Eurosoft Tech Private Limited, Pakistan  Full Time Jobs   
2                         ICM JAPAN, Pakistan  Full Time Jobs   
3                              ibex, Pakistan  Full Time Jobs   
4                         ICM JAPAN, Pakistan   Full Time Job   
...                                       ...             ...   
6675                       KnovaSol, Pakistan   Full Time Job   
6676                                      NaN   Full Time Job   
6677                  Loop Brackets, Pakistan   Full Time Job   
6678            Super Duper Studio , Pakistan  Full Time Jobs   
6679            Super Duper Studio , Pakistan  Full Time Jobs   

          Experience Required              Department  \
0            2 Years Job Exp.                 IT Jobs   
1            2 Years Job Exp.                 IT Jobs   
2                    < 1 Year   Customer Service Jobs   
3     Job for Fresh Graduates   Customer Service Jobs   
4                    < 1 Year    Customer Service Job   
...                       ...                     ...   
6675         3 Years Job Exp.   Computer Software Job   
6676         2 Years Job Exp.               Admin Job   
6677         2 Years Job Exp.   Computer Software Job   
6678         2 Years Job Exp.  Computer Software Jobs   
6679  Job for Fresh Graduates  Computer Software Jobs   

                                                     JD       City Date Posted  
0     New Job Positions .net, .netcore, flutter, Tea...  Islamabad   12-Mar-21  
1     We are looking for an experienced Web Develope...    Karachi   12-Mar-21  
2     International clients dealing exposure (B2B).S...    Karachi   12-Mar-21  
3     Responsible for acting as a liaison between ou...  Islamabad    9-Mar-21  
4     International clients dealing exposure (B2B) a...    Karachi    5-Mar-21  
...                                                 ...        ...         ...  
6675  We required services of Senior Software Engine...  Islamabad   30-Dec-20  
6676  As one of the leading employers in the country...     Multan   29-Dec-20  
6677  The ideal candidate will have experience in al...     Lahore   29-Dec-20  
6678  Must be able to create 3D Game Environments an...     Lahore   29-Dec-20  
6679  Candidate Shall Be Expert in Getting Work from...     Lahore   29-Dec-20  

[6680 rows x 9 columns]

**Number of Unique values in all columns**

In [46]:
for column in df:
    print("Column", "->",column,": Unique values are:", df[column].nunique())

Column -> Job Name : Unique values are: 3838
Column -> label : Unique values are: 3
Column -> Company Name : Unique values are: 2879
Column -> Job Type : Unique values are: 2
Column -> Experience Required : Unique values are: 17
Column -> Department : Unique values are: 111
Column -> JD : Unique values are: 5043
Column -> City : Unique values are: 86
Column -> Date Posted : Unique values are: 443


**Checking stats of City**

In [47]:
df['City']

0       Islamabad
1         Karachi
2         Karachi
3       Islamabad
4         Karachi
          ...    
6675    Islamabad
6676       Multan
6677       Lahore
6678       Lahore
6679       Lahore
Name: City, Length: 6680, dtype: object

**Unique and Number of unique value in this Column**

In [38]:
df['City'].unique()

array(['Islamabad', 'Karachi', 'Faisalabad', 'Lahore', 'Peshawar',
       'Sukkur', 'Talagang', 'Muridke', 'Dera Ghazi Khan', 'Rawalpindi',
       'Landi Kotal', 'Dera Ismail Khan', 'Jhang', 'Swat', 'Multan',
       'Sahiwal', 'Hyderabad', 'Mirpur', 'Chakwal', 'Shahdadpur', 'Sind',
       'Punjab', 'Sheikhupura', 'Shikarpur', 'Gujranwala', 'Daska',
       'Khushab', 'Lodhran', 'Dera Murad Jamali', 'Dir Lower', 'Ranipur',
       'Nawabshah', 'Mithi', 'Rahim Yar Khan', 'Attock', 'Bahawalpur',
       'Muzaffarabad', 'Balakot', 'Abbottabad', 'Sialkot', 'Malakand',
       'Bhakkar', 'Gujrat', 'Hasan Abdal', 'Mardan', 'Khanewal',
       'Ahmadpur East', 'Kharian', 'Mailsi', 'Sargodha', 'Haripur', 'Wah',
       'Bhalwal', 'Dipalpur', 'Ghakhar Mandi', 'Bahawalnagar', 'Pattoki',
       'Hazara', 'Bhurban', 'Ali Khan Abad', 'Chiniot', 'Besham',
       'Layyah', 'Muzaffargarh', 'Mirpur Khas', 'Gojra', 'Kasur', 'Badin',
       'Jhelum', 'Kamoke', 'Malakwal', 'Gujar Khan', 'Kamra', 'Quetta',
      

In [39]:
df['City'].nunique()

86

**Job only for Islamabad**

In [48]:
df[df["City"] == "Islamabad"] # How many Times Islamabad repeated

Job Name        label  \
0     Full Time New Job Positions .Net, .Netcore, Fl...  Premium Job   
3     Full Time Customer Support Specialist - Intern...  Premium Job   
6     Full Time Call Center - Lead Agent Jobs in Pak...  Premium Job   
11    Online Full Time Customer Sales Representative...      Hot Job   
16    Online Full Time Associate Software Engineer -...          NaN   
...                                                 ...          ...   
6660  Full Time Full Stack PHP / Wordpress Developer...          NaN   
6664  Full Time Bidding Executive / Business Develop...          NaN   
6668       Full Time Android Developer  Job in Pakistan          NaN   
6674  Full Time Senior Python Developer  Jobs in Pak...          NaN   
6675  Full Time Senior Software Engineer  Job in Pak...          NaN   

                                  Company Name        Job Type  \
0                    Nayel Solutions, Pakistan  Full Time Jobs   
3                               ibex, Pakistan  Full Time Jobs   
6            Pakistan Sweet Home AFP, Pakistan  Full Time Jobs   
11    Techative SMC(Private) Limited, Pakistan  Full Time Jobs   
16                                         NaN   Full Time Job   
...                                        ...             ...   
6660                      Allure SEO, Pakistan  Full Time Jobs   
6664                  MotifCreatives, Pakistan   Full Time Job   
6668                        KnovaSol, Pakistan   Full Time Job   
6674    Bridging Trade International, Pakistan  Full Time Jobs   
6675                        KnovaSol, Pakistan   Full Time Job   

          Experience Required             Department  \
0            2 Years Job Exp.                IT Jobs   
3     Job for Fresh Graduates  Customer Service Jobs   
6            3 Years Job Exp.             Sales Jobs   
11            1 Year Job Exp.  Customer Service Jobs   
16            1 Year Job Exp.        Engineering Job   
...                       ...                    ...   
6660         4 Years Job Exp.                IT Jobs   
6664         3 Years Job Exp.                 IT Job   
6668         2 Years Job Exp.  Computer Software Job   
6674         4 Years Job Exp.                IT Jobs   
6675         3 Years Job Exp.  Computer Software Job   

                                                     JD       City Date Posted  
0     New Job Positions .net, .netcore, flutter, Tea...  Islamabad   12-Mar-21  
3     Responsible for acting as a liaison between ou...  Islamabad    9-Mar-21  
6     Each Lead Agent is responsible for generating ...  Islamabad   22-Feb-21  
11    Techative SMC(Private) Limited is looking for ...  Islamabad   12-Mar-21  
16    What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21  
...                                                 ...        ...         ...  
6660  Allure SEO (www.allureseo.com) the Brooklyn-ba...  Islamabad   30-Dec-20  
6664  We are seeking talented, hardworking, &amp; mo...  Islamabad   30-Dec-20  
6668  We required services of Android developer.Mini...  Islamabad   30-Dec-20  
6674  What they will do:&middot; Development of appl...  Islamabad   30-Dec-20  
6675  We required services of Senior Software Engine...  Islamabad   30-Dec-20  

[1921 rows x 9 columns]

**Number of Duplicated values in Whole Dataset**

In [53]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
6675    False
6676     True
6677     True
6678     True
6679     True
Length: 6680, dtype: bool

In [52]:
df.duplicated().sum()

1055

In [54]:
df[df.duplicated() == True]

Job Name label  \
2503  Full Time 3D Modeler , CG Artist Game Jobs in ...   NaN   
5415     Full Time Assistant HR Manager Job in Pakistan   NaN   
5416  Online Full Time Associate Software Engineer -...   NaN   
5417  Full Time Associate Software Engineer - Fronte...   NaN   
5418  Online Full Time Associate Software Engineer -...   NaN   
...                                                 ...   ...   
6674  Full Time Senior Python Developer  Jobs in Pak...   NaN   
6676  Full Time Commercial Experience Executive Job ...   NaN   
6677  Full Time Business Development Executive Job i...   NaN   
6678  Full Time 3D Modeler / CG Artist Game Jobs in ...   NaN   
6679  Full Time Bidding Expert / Social Media Market...   NaN   

                                Company Name        Job Type  \
2503                    Syntax Bpo, Pakistan  Full Time Jobs   
5415                                     NaN   Full Time Job   
5416                                     NaN   Full Time Job   
5417                                     NaN   Full Time Job   
5418                                     NaN   Full Time Job   
...                                      ...             ...   
6674  Bridging Trade International, Pakistan  Full Time Jobs   
6676                                     NaN   Full Time Job   
6677                 Loop Brackets, Pakistan   Full Time Job   
6678           Super Duper Studio , Pakistan  Full Time Jobs   
6679           Super Duper Studio , Pakistan  Full Time Jobs   

          Experience Required              Department  \
2503         3 Years Job Exp.                 IT Jobs   
5415          1 Year Job Exp.   Computer Software Job   
5416          1 Year Job Exp.         Engineering Job   
5417          1 Year Job Exp.         Engineering Job   
5418          1 Year Job Exp.         Engineering Job   
...                       ...                     ...   
6674         4 Years Job Exp.                 IT Jobs   
6676         2 Years Job Exp.               Admin Job   
6677         2 Years Job Exp.   Computer Software Job   
6678         2 Years Job Exp.  Computer Software Jobs   
6679  Job for Fresh Graduates  Computer Software Jobs   

                                                     JD       City Date Posted  
2503  Must be able to create 3D Game Environments an...     Lahore   19-Sep-20  
5415  Looking for a go-getter, problem solving eager...     Lahore   15-Mar-21  
5416  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21  
5417  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21  
5418  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21  
...                                                 ...        ...         ...  
6674  What they will do:&middot; Development of appl...  Islamabad   30-Dec-20  
6676  As one of the leading employers in the country...     Multan   29-Dec-20  
6677  The ideal candidate will have experience in al...     Lahore   29-Dec-20  
6678  Must be able to create 3D Game Environments an...     Lahore   29-Dec-20  
6679  Candidate Shall Be Expert in Getting Work from...     Lahore   29-Dec-20  

[1055 rows x 9 columns]

**Removing Duplicate Values**

In [57]:
df.drop_duplicates()

Job Name        label  \
0     Full Time New Job Positions .Net, .Netcore, Fl...  Premium Job   
1       Full Time Senior Web Developer Jobs in Pakistan  Premium Job   
2           Full Time Russian Speakers Jobs in Pakistan  Premium Job   
3     Full Time Customer Support Specialist - Intern...  Premium Job   
4     Full Time English Speaker - International Busi...  Premium Job   
...                                                 ...          ...   
6665  Full Time Sr. Full Stack Engineer  Job in Paki...          NaN   
6666          Full Time Node Developer Jobs in Pakistan          NaN   
6667  Full Time Principal Software Engineer - Node J...          NaN   
6668       Full Time Android Developer  Job in Pakistan          NaN   
6675  Full Time Senior Software Engineer  Job in Pak...          NaN   

                                          Company Name        Job Type  \
0                            Nayel Solutions, Pakistan  Full Time Jobs   
1              Eurosoft Tech Private Limited, Pakistan  Full Time Jobs   
2                                  ICM JAPAN, Pakistan  Full Time Jobs   
3                                       ibex, Pakistan  Full Time Jobs   
4                                  ICM JAPAN, Pakistan   Full Time Job   
...                                                ...             ...   
6665  Professional Employers Private Limited, Pakistan   Full Time Job   
6666  Professional Employers Private Limited, Pakistan  Full Time Jobs   
6667  Professional Employers Private Limited, Pakistan   Full Time Job   
6668                                KnovaSol, Pakistan   Full Time Job   
6675                                KnovaSol, Pakistan   Full Time Job   

          Experience Required              Department  \
0            2 Years Job Exp.                 IT Jobs   
1            2 Years Job Exp.                 IT Jobs   
2                    < 1 Year   Customer Service Jobs   
3     Job for Fresh Graduates   Customer Service Jobs   
4                    < 1 Year    Customer Service Job   
...                       ...                     ...   
6665         2 Years Job Exp.   Computer Software Job   
6666         2 Years Job Exp.  Computer Software Jobs   
6667         4 Years Job Exp.   Computer Software Job   
6668         2 Years Job Exp.   Computer Software Job   
6675         3 Years Job Exp.   Computer Software Job   

                                                     JD       City Date Posted  
0     New Job Positions .net, .netcore, flutter, Tea...  Islamabad   12-Mar-21  
1     We are looking for an experienced Web Develope...    Karachi   12-Mar-21  
2     International clients dealing exposure (B2B).S...    Karachi   12-Mar-21  
3     Responsible for acting as a liaison between ou...  Islamabad    9-Mar-21  
4     International clients dealing exposure (B2B) a...    Karachi    5-Mar-21  
...                                                 ...        ...         ...  
6665  Job description:Adapting interface for modern ...     Lahore   30-Dec-20  
6666  We are looking for a highly capable Node.js se...     Lahore   30-Dec-20  
6667  We are looking for a highly capable Node.js se...     Lahore   30-Dec-20  
6668  We required services of Android developer.Mini...  Islamabad   30-Dec-20  
6675  We required services of Senior Software Engine...  Islamabad   30-Dec-20  

[5625 rows x 9 columns]

### Q4 Utilize str.strip() to remove leading and trailing spaces and str.replace() function. 

**Columns Name in DataSet**

In [63]:
df.columns

Index(['Job Name', 'label', 'Company Name', 'Job Type', 'Experience Required',
       'Department', 'JD', 'City', 'Date Posted'],
      dtype='object')

**Removes leading and trailing spaces**

In [62]:
df.columns.str.strip()

Index(['Job Name', 'label', 'Company Name', 'Job Type', 'Experience Required',
       'Department', 'JD', 'City', 'Date Posted'],
      dtype='object')

**Replaces the name of column from Label to Title**

In [64]:
df.columns.str.replace('label','Title')

Index(['Job Name', 'Title', 'Company Name', 'Job Type', 'Experience Required',
       'Department', 'JD', 'City', 'Date Posted'],
      dtype='object')

In [65]:
df.columns.str.replace('Title','label')

Index(['Job Name', 'label', 'Company Name', 'Job Type', 'Experience Required',
       'Department', 'JD', 'City', 'Date Posted'],
      dtype='object')

### Q5 Use loc & iloc in four different codes and print the output.

**iloc function prints values starting from the first input and ending at the second input but not including it.**

In [71]:
df.iloc[10:20]

Job Name    label  \
10   Full Time Sr. Backend Developer Jobs in Pakistan  Hot Job   
11  Online Full Time Customer Sales Representative...  Hot Job   
12  Full Time Client Relationship Executive Jobs i...  Hot Job   
13  Full Time Business Development Executive - BDE...  Hot Job   
14  Full Time BDE - Business Development Executive...  Hot Job   
15     Full Time Assistant HR Manager Job in Pakistan      NaN   
16  Online Full Time Associate Software Engineer -...      NaN   
17  Full Time Associate Software Engineer - Fronte...      NaN   
18  Online Full Time Associate Software Engineer -...      NaN   
19      Full Time Executive Assistant Job in Pakistan      NaN   

                                Company Name        Job Type  \
10         Magnus Mage (Pvt.) Ltd., Pakistan  Full Time Jobs   
11  Techative SMC(Private) Limited, Pakistan  Full Time Jobs   
12              HG Markets Pvt Ltd, Pakistan  Full Time Jobs   
13              HG Markets Pvt Ltd, Pakistan  Full Time Jobs   
14              HG Markets Pvt Ltd, Pakistan  Full Time Jobs   
15                                       NaN   Full Time Job   
16                                       NaN   Full Time Job   
17                                       NaN   Full Time Job   
18                                       NaN   Full Time Job   
19                                       NaN   Full Time Job   

   Experience Required              Department  \
10     1 Year Job Exp.  Computer Software Jobs   
11     1 Year Job Exp.   Customer Service Jobs   
12            < 1 Year              Sales Jobs   
13            < 1 Year              Sales Jobs   
14            < 1 Year              Sales Jobs   
15     1 Year Job Exp.   Computer Software Job   
16     1 Year Job Exp.         Engineering Job   
17     1 Year Job Exp.         Engineering Job   
18     1 Year Job Exp.         Engineering Job   
19     1 Year Job Exp.               Admin Job   

                                                   JD       City Date Posted  
10  We are looking for Backend Developers (NodeJS/...     Lahore   15-Mar-21  
11  Techative SMC(Private) Limited is looking for ...  Islamabad   12-Mar-21  
12  To generate quality leads that result in sales...     Lahore   12-Mar-21  
13  To generate quality leads that result in sales...     Lahore   12-Mar-21  
14  To generate quality leads that result in sales...     Lahore    5-Mar-21  
15  Looking for a go-getter, problem solving eager...     Lahore   15-Mar-21  
16  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21  
17  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21  
18  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21  
19  Remotebase is looking for our C-suite’s right-...     Lahore   15-Mar-21

**The inputs after comma will print the specific column staring from first and at the second input but not including it**

In [70]:
df.iloc[10:20,0:3]

Job Name    label  \
10   Full Time Sr. Backend Developer Jobs in Pakistan  Hot Job   
11  Online Full Time Customer Sales Representative...  Hot Job   
12  Full Time Client Relationship Executive Jobs i...  Hot Job   
13  Full Time Business Development Executive - BDE...  Hot Job   
14  Full Time BDE - Business Development Executive...  Hot Job   
15     Full Time Assistant HR Manager Job in Pakistan      NaN   
16  Online Full Time Associate Software Engineer -...      NaN   
17  Full Time Associate Software Engineer - Fronte...      NaN   
18  Online Full Time Associate Software Engineer -...      NaN   
19      Full Time Executive Assistant Job in Pakistan      NaN   

                                Company Name  
10         Magnus Mage (Pvt.) Ltd., Pakistan  
11  Techative SMC(Private) Limited, Pakistan  
12              HG Markets Pvt Ltd, Pakistan  
13              HG Markets Pvt Ltd, Pakistan  
14              HG Markets Pvt Ltd, Pakistan  
15                                       NaN  
16                                       NaN  
17                                       NaN  
18                                       NaN  
19                                       NaN

**loc function prints values starting from the first input and ending at the second input including it.**

In [143]:
df.loc[10:20]

Job Name    label  \
10   Full Time Sr. Backend Developer Jobs in Pakistan  Hot Job   
11  Online Full Time Customer Sales Representative...  Hot Job   
12  Full Time Client Relationship Executive Jobs i...  Hot Job   
13  Full Time Business Development Executive - BDE...  Hot Job   
14  Full Time BDE - Business Development Executive...  Hot Job   
15     Full Time Assistant HR Manager Job in Pakistan      NaN   
16  Online Full Time Associate Software Engineer -...      NaN   
17  Full Time Associate Software Engineer - Fronte...      NaN   
18  Online Full Time Associate Software Engineer -...      NaN   
19      Full Time Executive Assistant Job in Pakistan      NaN   
20  Full Time Maketing and PR Specialist Job in Pa...      NaN   

                                Company Name        Job Type  \
10         Magnus Mage (Pvt.) Ltd., Pakistan  Full Time Jobs   
11  Techative SMC(Private) Limited, Pakistan  Full Time Jobs   
12              HG Markets Pvt Ltd, Pakistan  Full Time Jobs   
13              HG Markets Pvt Ltd, Pakistan  Full Time Jobs   
14              HG Markets Pvt Ltd, Pakistan  Full Time Jobs   
15                                       NaN   Full Time Job   
16                                       NaN   Full Time Job   
17                                       NaN   Full Time Job   
18                                       NaN   Full Time Job   
19                                       NaN   Full Time Job   
20                                       NaN   Full Time Job   

   Experience Required              Department  \
10     1 Year Job Exp.  Computer Software Jobs   
11     1 Year Job Exp.   Customer Service Jobs   
12            < 1 Year              Sales Jobs   
13            < 1 Year              Sales Jobs   
14            < 1 Year              Sales Jobs   
15     1 Year Job Exp.   Computer Software Job   
16     1 Year Job Exp.         Engineering Job   
17     1 Year Job Exp.         Engineering Job   
18     1 Year Job Exp.         Engineering Job   
19     1 Year Job Exp.               Admin Job   
20     1 Year Job Exp.   Computer Software Job   

                                                   JD       City Date Posted  \
10  We are looking for Backend Developers (NodeJS/...     Lahore   15-Mar-21   
11  Techative SMC(Private) Limited is looking for ...  Islamabad   12-Mar-21   
12  To generate quality leads that result in sales...     Lahore   12-Mar-21   
13  To generate quality leads that result in sales...     Lahore   12-Mar-21   
14  To generate quality leads that result in sales...     Lahore    5-Mar-21   
15  Looking for a go-getter, problem solving eager...     Lahore   15-Mar-21   
16  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21   
17  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21   
18  What You'll Do:Write, test, and document code....  Islamabad   15-Mar-21   
19  Remotebase is looking for our C-suite’s right-...     Lahore   15-Mar-21   
20  Become a part of one of the fastest growing st...     Lahore   15-Mar-21   

    Salary Employee  
10           160000  
11           160000  
12           160000  
13           160000  
14           160000  
15           160000  
16           160000  
17           160000  
18           160000  
19           160000  
20           160000

**You can print specific column using loc fucntion using the name of column**

In [144]:
df.loc[10:20,["City"]]

City
10     Lahore
11  Islamabad
12     Lahore
13     Lahore
14     Lahore
15     Lahore
16  Islamabad
17  Islamabad
18  Islamabad
19     Lahore
20     Lahore

### Q6 Determine the missing values in the whole dataset and analyze missing values in each column. 

**Number of Null values in each column**

In [145]:
df.isna().sum()

Job Name                  0
label                  5565
Company Name            663
Job Type                  0
Experience Required       0
Department                0
JD                        0
City                      0
Date Posted               0
Salary Employee           0
dtype: int64

**Number of Null values in label and Company Name column**

In [146]:
df['label'].isna().sum()

5565

In [147]:
df['Company Name'].isna().sum()

663

### Q7 Calculate the sum of duplicate rows.

**Number of Duplicated values in Whole Dataset**

In [148]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
6675    False
6676     True
6677     True
6678     True
6679     True
Length: 6680, dtype: bool

In [149]:
df.duplicated().sum()

1055

### Q8. Use conditional statements and solve multiple questions a minimum of five.

**1 if City is islamabad and experince is less then 1 year**

In [150]:
df[(df["City"]== 'Islamabad') & (df['Experience Required'] == "< 1 Year")]

Job Name        label  \
30    Full Time Pharmacist With Category Required  J...  Gallery Job   
201   Full Time CBS Rapid - Administrative And Workp...          NaN   
376   Full Time Art Gallery Coordinator Job in Pakistan          NaN   
400   Full Time CSR, Closers, Team Leaders Jobs in P...          NaN   
474   Full Time Senior Web / Graphic Designer Job in...          NaN   
...                                                 ...          ...   
6559  Full Time Customer Service Representative Jobs...          NaN   
6597  Full Time  Developer (.NET Core) Jobs in Pakistan          NaN   
6608         Full Time Biology Teacher  Job in Pakistan          NaN   
6638  Full Time Customer Relationship Officer Jobs i...          NaN   
6656  Full Time Mechanical Engineer Site Job in Paki...          NaN   

                                          Company Name        Job Type  \
30                            Talat Pharmacy, Pakistan   Full Time Job   
201                                                NaN   Full Time Job   
376                  Xerosector Technologies, Pakistan   Full Time Job   
400                 Truss Communications BPO, Pakistan  Full Time Jobs   
474   Reach First Communications SMC Pvt Ltd, Pakistan   Full Time Job   
...                                                ...             ...   
6559                     Divergent Solutions, Pakistan  Full Time Jobs   
6597                          Futuresol Tech, Pakistan  Full Time Jobs   
6608        Asim Foundation School & Academy, Pakistan   Full Time Job   
6638             Starland Marketing Pvt Ltd., Pakistan  Full Time Jobs   
6656    Innovative Communications (Pvt) Ltd., Pakistan   Full Time Job   

     Experience Required             Department  \
30              < 1 Year           Pharmacy Job   
201             < 1 Year         Accounting Job   
376             < 1 Year             Retail Job   
400             < 1 Year  Customer Service Jobs   
474             < 1 Year          Marketing Job   
...                  ...                    ...   
6559            < 1 Year  Customer Service Jobs   
6597            < 1 Year                IT Jobs   
6608            < 1 Year          Education Job   
6638            < 1 Year  Customer Service Jobs   
6656            < 1 Year       Construction Job   

                                                     JD       City  \
30    pharmacist with category required for phy in d...  Islamabad   
201   Administrative &amp; Workplace Services Intern...  Islamabad   
376   A newly established ART Gallery in Bahria town...  Islamabad   
400   Truss Communications BPO is hiring competitive...  Islamabad   
474   Company Overview:Reach First is a one-stop dig...  Islamabad   
...                                                 ...        ...   
6559  Looking for fresh and experience agent for Sol...  Islamabad   
6597  We are looking for .NET Core and MVC (develope...  Islamabad   
6608  Asim Foundation School Gulraiz branch (Rawalpi...  Islamabad   
6638  We are looking for Female Customer services Re...  Islamabad   
6656  Assist to Senior mecchnical DPM at sie forsett...  Islamabad   

     Date Posted  Salary Employee  
30     13-Mar-21           160000  
201    31-Jan-21           160000  
376    12-Mar-21           160000  
400    10-Mar-21           160000  
474     3-Mar-21           160000  
...          ...              ...  
6559   10-Jan-21           160000  
6597    6-Jan-21           160000  
6608    6-Jan-21           160000  
6638    2-Jan-21           160000  
6656   31-Dec-20           160000  

[412 rows x 10 columns]

**2 if City is islamabad and experince is less then 1 year and Job type fulltime**

In [151]:
df[(df["City"]== 'Islamabad') & (df['Experience Required'] == "< 1 Year") & (df['Job Type'] == "Full Time Job")]

Job Name        label  \
30    Full Time Pharmacist With Category Required  J...  Gallery Job   
201   Full Time CBS Rapid - Administrative And Workp...          NaN   
376   Full Time Art Gallery Coordinator Job in Pakistan          NaN   
474   Full Time Senior Web / Graphic Designer Job in...          NaN   
475              Full Time JS Developer Job in Pakistan          NaN   
...                                                 ...          ...   
6463  Full Time Front End / Wordpress / Shopify / Wi...          NaN   
6487  Online Full Time Content Writer, Copywriter - ...          NaN   
6514  Full Time Assistant Manager Marketing & Sales ...          NaN   
6608         Full Time Biology Teacher  Job in Pakistan          NaN   
6656  Full Time Mechanical Engineer Site Job in Paki...          NaN   

                                          Company Name       Job Type  \
30                            Talat Pharmacy, Pakistan  Full Time Job   
201                                                NaN  Full Time Job   
376                  Xerosector Technologies, Pakistan  Full Time Job   
474   Reach First Communications SMC Pvt Ltd, Pakistan  Full Time Job   
475   Reach First Communications SMC Pvt Ltd, Pakistan  Full Time Job   
...                                                ...            ...   
6463                               InnovaSol, Pakistan  Full Time Job   
6487                            Ecom Digital, Pakistan  Full Time Job   
6514                          IFA Cambridge , Pakistan  Full Time Job   
6608        Asim Foundation School & Academy, Pakistan  Full Time Job   
6656    Innovative Communications (Pvt) Ltd., Pakistan  Full Time Job   

     Experience Required            Department  \
30              < 1 Year          Pharmacy Job   
201             < 1 Year        Accounting Job   
376             < 1 Year            Retail Job   
474             < 1 Year         Marketing Job   
475             < 1 Year         Marketing Job   
...                  ...                   ...   
6463            < 1 Year  Web & E-commerce Job   
6487            < 1 Year         Marketing Job   
6514            < 1 Year         Marketing Job   
6608            < 1 Year         Education Job   
6656            < 1 Year      Construction Job   

                                                     JD       City  \
30    pharmacist with category required for phy in d...  Islamabad   
201   Administrative &amp; Workplace Services Intern...  Islamabad   
376   A newly established ART Gallery in Bahria town...  Islamabad   
474   Company Overview:Reach First is a one-stop dig...  Islamabad   
475   Company Overview:Reach First is a one-stop dig...  Islamabad   
...                                                 ...        ...   
6463  InnovSol are looking web developer for company...  Islamabad   
6487  This is a remote job.Looking for Content Write...  Islamabad   
6514  Need candidates who have passion in marketing ...  Islamabad   
6608  Asim Foundation School Gulraiz branch (Rawalpi...  Islamabad   
6656  Assist to Senior mecchnical DPM at sie forsett...  Islamabad   

     Date Posted  Salary Employee  
30     13-Mar-21           160000  
201    31-Jan-21           160000  
376    12-Mar-21           160000  
474     3-Mar-21           160000  
475     3-Mar-21           160000  
...          ...              ...  
6463   18-Jan-21           160000  
6487   15-Jan-21           160000  
6514   13-Jan-21           160000  
6608    6-Jan-21           160000  
6656   31-Dec-20           160000  

[179 rows x 10 columns]

**3 if City is Karachi and experince is less then 1 year and Departement Marketing**

In [152]:
df[(df["City"]== 'Karachi') & (df['Experience Required'] == "< 1 Year") & (df['Department'] == "Marketing Job")]

Job Name        label  \
382   Full Time E Commerce Executive / Assistant Job...          NaN   
447     Full Time Animator (Internship) Job in Pakistan          NaN   
540          Full Time Digital Marketer Job in Pakistan          NaN   
967         Full Time Graphics Designer Job in Pakistan          NaN   
1497     Full Time Affiliate Marketing  Job in Pakistan          NaN   
1707  Full Time SEO Expert And Web Developer Job in ...          NaN   
1708  Full Time Social Media Marketing And Graphic D...          NaN   
1910           Full Time Vector Artist  Job in Pakistan          NaN   
2543     Full Time 3D Graphics Designer Job in Pakistan  Gallery Job   
2551  Online Full Time Digital Media Marketing & Sal...          NaN   
2840         Full Time Digital Marketer Job in Pakistan          NaN   
2924            Full Time Web Developer Job in Pakistan          NaN   
3334  Online Full Time Digital Marketing Executive  ...      Hot Job   
3446      Full Time Marketing Assistant Job in Pakistan          NaN   
3452        Full Time Marketing Officer Job in Pakistan          NaN   
4090  Full Time Graphic Designer And Digital Markete...          NaN   
4661  Full Time RUSSIAN SPEAKERS - Int. Business Dev...  Premium Job   
4663  Full Time Int. Business Development Executive ...  Premium Job   
4818   Full Time Social Media Executive Job in Pakistan          NaN   
5782  Full Time E Commerce Executive / Assistant Job...          NaN   
5847    Full Time Animator (Internship) Job in Pakistan          NaN   
5940         Full Time Digital Marketer Job in Pakistan          NaN   
6367        Full Time Graphics Designer Job in Pakistan          NaN   

                                           Company Name       Job Type  \
382                        Nazreen Consulting, Pakistan  Full Time Job   
447                                 Use Group, Pakistan  Full Time Job   
540   Pulp Consultants (SMC - Private) Limited, Paki...  Full Time Job   
967   Crown Q Institution of Hospitality Management,...  Full Time Job   
1497                        Digital Cloud Hub, Pakistan  Full Time Job   
1707                Exact Management Services, Pakistan  Full Time Job   
1708                Exact Management Services, Pakistan  Full Time Job   
1910                                Red Coral, Pakistan  Full Time Job   
2543                            Creative Arts, Pakistan  Full Time Job   
2551                         MS Tech Services, Pakistan  Full Time Job   
2840              Big Four Experts Accounting, Pakistan  Full Time Job   
2924                       Aesthetics Digital, Pakistan  Full Time Job   
3334                             Burger King , Pakistan  Full Time Job   
3446                           Kaaf Engineers, Pakistan  Full Time Job   
3452                           Kaaf Engineers, Pakistan  Full Time Job   
4090                         Innovative Ducks, Pakistan  Full Time Job   
4661                                ICM JAPAN, Pakistan  Full Time Job   
4663                                ICM JAPAN, Pakistan  Full Time Job   
4818                 AppedologyPk Recruitment, Pakistan  Full Time Job   
5782                       Nazreen Consulting, Pakistan  Full Time Job   
5847                                Use Group, Pakistan  Full Time Job   
5940  Pulp Consultants (SMC - Private) Limited, Paki...  Full Time Job   
6367  Crown Q Institution of Hospitality Management,...  Full Time Job   

     Experience Required     Department  \
382             < 1 Year  Marketing Job   
447             < 1 Year  Marketing Job   
540             < 1 Year  Marketing Job   
967             < 1 Year  Marketing Job   
1497            < 1 Year  Marketing Job   
1707            < 1 Year  Marketing Job   
1708            < 1 Year  Marketing Job   
1910            < 1 Year  Marketing Job   
2543            < 1 Year  Marketing Job   
2551            < 1 Year  Marketing Job   
2840            < 1 Year  Marketing Job   
2924            < 1 Year 

**4 if City is islamabad or lahore and Job type fulltime**

In [153]:
df[(df["City"]== 'Islamabad') | (df["City"]== 'Lahore') & (df['Job Type'] == "Full Time Job")]

Job Name        label  \
0     Full Time New Job Positions .Net, .Netcore, Fl...  Premium Job   
3     Full Time Customer Support Specialist - Intern...  Premium Job   
6     Full Time Call Center - Lead Agent Jobs in Pak...  Premium Job   
11    Online Full Time Customer Sales Representative...      Hot Job   
15       Full Time Assistant HR Manager Job in Pakistan          NaN   
...                                                 ...          ...   
6667  Full Time Principal Software Engineer - Node J...          NaN   
6668       Full Time Android Developer  Job in Pakistan          NaN   
6674  Full Time Senior Python Developer  Jobs in Pak...          NaN   
6675  Full Time Senior Software Engineer  Job in Pak...          NaN   
6677  Full Time Business Development Executive Job i...          NaN   

                                          Company Name        Job Type  \
0                            Nayel Solutions, Pakistan  Full Time Jobs   
3                                       ibex, Pakistan  Full Time Jobs   
6                    Pakistan Sweet Home AFP, Pakistan  Full Time Jobs   
11            Techative SMC(Private) Limited, Pakistan  Full Time Jobs   
15                                                 NaN   Full Time Job   
...                                                ...             ...   
6667  Professional Employers Private Limited, Pakistan   Full Time Job   
6668                                KnovaSol, Pakistan   Full Time Job   
6674            Bridging Trade International, Pakistan  Full Time Jobs   
6675                                KnovaSol, Pakistan   Full Time Job   
6677                           Loop Brackets, Pakistan   Full Time Job   

          Experience Required             Department  \
0            2 Years Job Exp.                IT Jobs   
3     Job for Fresh Graduates  Customer Service Jobs   
6            3 Years Job Exp.             Sales Jobs   
11            1 Year Job Exp.  Customer Service Jobs   
15            1 Year Job Exp.  Computer Software Job   
...                       ...                    ...   
6667         4 Years Job Exp.  Computer Software Job   
6668         2 Years Job Exp.  Computer Software Job   
6674         4 Years Job Exp.                IT Jobs   
6675         3 Years Job Exp.  Computer Software Job   
6677         2 Years Job Exp.  Computer Software Job   

                                                     JD       City  \
0     New Job Positions .net, .netcore, flutter, Tea...  Islamabad   
3     Responsible for acting as a liaison between ou...  Islamabad   
6     Each Lead Agent is responsible for generating ...  Islamabad   
11    Techative SMC(Private) Limited is looking for ...  Islamabad   
15    Looking for a go-getter, problem solving eager...     Lahore   
...                                                 ...        ...   
6667  We are looking for a highly capable Node.js se...     Lahore   
6668  We required services of Android developer.Mini...  Islamabad   
6674  What they will do:&middot; Development of appl...  Islamabad   
6675  We required services of Senior Software Engine...  Islamabad   
6677  The ideal candidate will have experience in al...     Lahore   

     Date Posted  Salary Employee  
0      12-Mar-21           160000  
3       9-Mar-21           160000  
6      22-Feb-21           160000  
11     12-Mar-21           160000  
15     15-Mar-21           160000  
...          ...              ...  
6667   30-Dec-20           160000  
6668   30-Dec-20           160000  
6674   30-Dec-20           160000  
6675   30-Dec-20           160000  
6677   29-Dec-20           160000  

[3053 rows x 10 columns]

**5 if Company name is ibex,  Experience is Fresh Graduates**

In [154]:
df[(df["Company Name"]== 'ibex, Pakistan') & (df['Experience Required'] == "Job for Fresh Graduates")]

Job Name        label  \
3     Full Time Customer Support Specialist - Intern...  Premium Job   
404   Full Time Email And Chat Support Executive - F...  Gallery Job   
405   Full Time Email/Chat Support Specialist - Reta...  Gallery Job   
479   Full Time Inbound Sales Specialist Jobs in Pak...  Gallery Job   
481   Full Time Inbound Sales Executive  Jobs in Pak...  Gallery Job   
4013  Full Time Medical Representative Jobs in Pakistan  Gallery Job   
4054  Full Time Customer Support Specialist - Intern...  Gallery Job   
4064     Online Full Time Image Editor Jobs in Pakistan  Premium Job   
4079          Full Time Pashto Speaker Jobs in Pakistan  Gallery Job   
4139  Full Time Japanese Speaker - International  Jo...  Gallery Job   
4141  Full Time Chinese Speaker - International Jobs...  Gallery Job   
4142  Full Time Arabic - Email/ Chat Support Executi...  Gallery Job   
4143  Full Time Inbound Sales Specialist - Karachi J...  Gallery Job   
5403  Full Time Customer Support Specialist - Intern...          NaN   
5804  Full Time Email And Chat Support Executive - F...          NaN   
5805  Full Time Email/Chat Support Specialist - Reta...          NaN   
5879  Full Time Inbound Sales Specialist Jobs in Pak...          NaN   
5881  Full Time Inbound Sales Executive  Jobs in Pak...          NaN   

        Company Name        Job Type      Experience Required  \
3     ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
404   ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
405   ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
479   ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
481   ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4013  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4054  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4064  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4079  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4139  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4141  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4142  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
4143  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
5403  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
5804  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
5805  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
5879  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   
5881  ibex, Pakistan  Full Time Jobs  Job for Fresh Graduates   

                 Department  \
3     Customer Service Jobs   
404   Customer Service Jobs   
405   Customer Service Jobs   
479   Customer Service Jobs   
481   Customer Service Jobs   
4013  Customer Service Jobs   
4054  Customer Service Jobs   
4064  Customer Service Jobs   
4079  Customer Service Jobs   
4139  Customer Service Jobs   
4141  Customer Service Jobs   
4142  Customer Service Jobs   
4143  Customer Service Jobs   
5403  Customer Service Jobs   
5804  Customer Service Jobs   
5805  Customer Service Jobs   
5879  Customer Service Jobs   
5881  Customer Service Jobs   

                                                     JD       City  \
3     Responsible for acting as a liaison between ou...  Islamabad   
404   Responsible for acting as a liaison between ou...    Karachi   
405   Location : KarachiJob Overview:\nEmail/Chat Su...    Karachi   
479   An Inbound Sales Specialist plays an important...    Karachi   
481   Job Description Answer inbound calls from pros...     Lahore   
4013  Job Overview: Provide medical or treatment inf...    Karachi   
4054  Job Description:Assist customers in resolving ...     Lahore   
4064  Location: KarachiSalary: Up to PKR 40,000Worki...    Karachi   
4079  Job Overview: Responsible for acting as a liai...    Karachi   
4139  Interact with customers to provide and process...    Karachi   
4141  Collaborate and build relationships with custo...    Karachi   
4142  Intera

### Q9  Apply the cut & qcut technique and compare the results.

My Dataset does't have the Integer values I am creating a small data set for Qcut and cut

In [155]:
data =  [["Alice",10],["Bob",20],["Charlie",30],["David",40],["Eva",50],["Frank",60],["Grace",70],["Hannah",80],["Ivy",90],["Jack",100]]

In [156]:
df1 = pd.DataFrame(data, columns=['Name', 'Age'])
df1

Name  Age
0    Alice   10
1      Bob   20
2  Charlie   30
3    David   40
4      Eva   50
5    Frank   60
6    Grace   70
7   Hannah   80
8      Ivy   90
9     Jack  100

In [157]:
pd.qcut(df1["Age"],q=4)

0    (9.999, 32.5]
1    (9.999, 32.5]
2    (9.999, 32.5]
3     (32.5, 55.0]
4     (32.5, 55.0]
5     (55.0, 77.5]
6     (55.0, 77.5]
7    (77.5, 100.0]
8    (77.5, 100.0]
9    (77.5, 100.0]
Name: Age, dtype: category
Categories (4, interval[float64, right]): [(9.999, 32.5] < (32.5, 55.0] < (55.0, 77.5] < (77.5, 100.0]]

**Qcut**

In [158]:
binname = ['young', 'mid young', 'middle ', 'experience']
df1['Age range'] = pd.qcut(df1["Age"],q=4,labels=binname)

In [159]:
df1['Age range']

0         young
1         young
2         young
3     mid young
4     mid young
5       middle 
6       middle 
7    experience
8    experience
9    experience
Name: Age range, dtype: category
Categories (4, object): ['young' < 'mid young' < 'middle ' < 'experience']

**Cut**

In [160]:
pd.cut(df1["Age"],bins=4,labels = ['Teen','Young','Middle',"Old"])

0      Teen
1      Teen
2      Teen
3     Young
4     Young
5    Middle
6    Middle
7       Old
8       Old
9       Old
Name: Age, dtype: category
Categories (4, object): ['Teen' < 'Young' < 'Middle' < 'Old']